app/assemblers/assemble_tasks.rb

```diff
@@ -30,6 +30,6 @@ def tasks
 
   memoize
   def track_id
-    Track.find(params[:track]).id if params[:track].present?
+    Track.find_by!(slug: params[:track_slug]).id if params[:track_slug].present?
```

Comment:
I think this should remain as `Track.find` rather than `Track.find_by` as track uses friendly-id to handle renamed things.

If you wanted to be explicit that these take a slug, you could change them to `Track.friendly.find(...)` instead

```suggestion
    Track.find(params[:track_slug]).id if params[:track_slug].present?
```

Reply:
Cool. I will do this.